### ISSUE TITLE
init container needs imagePullPolicy in kubernetes 1.5

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/fabric8-maven-plugin#734]
if we use fabric8 on kubernetes 1.5 we get these errors:
```
Failed to create Deployment: Deployment.extensions "nexus" is invalid: spec.template.spec.initContainers[0].imagePullPolicy: Required valueplatform......................................................................✘ Deployment.extensions "nexus" is invalid: spec.template.spec.initContainers[0].imagePullPolicy: Required value
Failed to create Deployment: Deployment.extensions "gogs" is invalid: spec.template.spec.initContainers[0].imagePullPolicy: Required valueplatform......................................................................✘ Deployment.extensions "gogs" is invalid: spec.template.spec.initContainers[0].imagePullPolicy: Required value
```

it looks like we need to add an `imagePullPolicy` now

**Comment by jstrachan** [comment]
looks like its a Kubernetes bug - kubernetes/kubernetes#38542 with potential fix in kubernetes/kubernetes#38574

so we maybe don't need to fix it? Though it would not be hard to default a value

**Comment by novakg** [comment]
Hi,
I'm using Kubernetes v1.5.1 and experiencing the same issue.
Do you know a relatively easy workaround that does not require recompiling the whole Kubernetes? I mean something like editing the necessary yaml files of nexus, gogs and jenkins services in a separate branch and manually feeding them into kubernetes.

**Comment by jstrachan** [comment]
we're working as fast as we can to get a fabric8 release out; should be there soon.

Until then the best workaround is to stick with 1.4.x of kubernetes (e.g. use an older minikube)

**Comment by sunildurai** [comment]
Thanks for the response; Do you have any update on the fix?

**Comment by novakg** [comment]
As an additional information: I could not stick with 1.4.x of Kubernetes as jstrachan suggested (as I needed the internal IP based communication of nodes for OpenStack, which has been introduced in Kubernetes 1.5), so I've rebuilt Kubernetes 1.5.1 with the changes of [cd6792ae0879b2fbe831e394b0831433f068a420](https://github.com/kubernetes/kubernetes/pull/38574/commits/cd6792ae0879b2fbe831e394b0831433f068a420) instead and it works fine now (with Fabric8 v0.4.105).

**Comment by jstrachan** [comment]
this is now fixed in 2.4.15 of fabric8-platform
